src/libsyntax/parse/mod.rs

```diff
@@ -318,7 +310,10 @@ pub fn char_lit(lit: &str) -> (char, isize) {
 
     let unicode_escape = || -> Option<(char, isize)> {
         if lit.as_bytes()[2] == b'{' {
-            let idx = lit.find('}').expect(msg2);
+            let idx = match lit.find('}') {
+                Some(idx2) => idx2,
+                None => panic!("lexer should have rejected a bad character escape {}", lit),
+            };
```

Comment:
That's true, but you're overlooking the speed improvement which is the motivation for the change.